java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java

```diff
@@ -321,12 +345,15 @@ public void write(Iterator<SinkRow> rows) {
                 throw new RuntimeException(ex);
             }
         }
+        checkWriterError();
     }
 
     @Override
     public void sync() {
         try {
-            bulkProcessor.flush();
+            bulkProcessor.awaitClose(1, TimeUnit.SECONDS);
```

Comment:
hmm... Why do we close and recreate the `bulkProcessor` here? Will it cause the in-flight batch to be cancelled even though no errors occur?

Reply:
Because the flush method will only ensure the execution of the feature in the queue, will not ensure his completion, and we need to ensure the completion of the method in a barrier, and will throw the error, So we use waitClose method, he will ensure that so the task is completed after waiting for the closure.

Reply:
It doesn't make sense for us to fail and recover when the bulkProcessor fails to flush all requests in 1 second.